crucible-llvm/src/Lang/Crucible/LLVM/Intrinsics/Libc.hs

```diff
@@ -487,6 +487,20 @@ callAssert mvar sym (Empty :> _pfn :> _pfile :> _pline :> ptxt ) =
         let err = AssertFailureSimError "Call to assert()" (UTF8.toString txt)
         liftIO $ addFailedAssertion sym err
 
+callExit :: (IsSymInterface sym)
+         => sym
+         -> RegEntry sym (BVType 32)
+         -> OverrideSim p sym ext r args ret (RegValue sym UnitType)
+callExit sym ec = liftIO $
```

Comment:
This is an odd pattern.  I think I'd rather see an assertion that the code is zero, followed by an `abortExecBecause`.

Reply:
Ah, good suggestion. As an added bonus, this no longer exhibits the weird unreachable code error seen in https://github.com/GaloisInc/crucible/pull/786#discussion_r675112827.